receiver/influxdbreceiver/receiver.go

```diff
@@ -201,6 +212,23 @@ func (r *metricsReceiver) handleWrite(w http.ResponseWriter, req *http.Request)
 		return
 	}
 
+	if len(failureReason) > 0 {
+		totalErrors := 0
+		for _, count := range failureReason {
+			totalErrors += count
+		}
+		w.WriteHeader(http.StatusBadRequest)
+		w.Header().Set("Content-Type", "application/json")
```

Comment:
Setting headers should come before `.WriteHeader`.
```suggestion
		w.Header().Set("Content-Type", "application/json")
		w.WriteHeader(http.StatusBadRequest)
```